test/e2e/snaps/test-snap-onoff.spec.js

```diff
@@ -0,0 +1,155 @@
+        windowHandles = await driver.getAllWindowHandles();
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement({
+          text: 'Approve & install',
+          tag: 'button',
+        });
+
+        // switch to the original MM tab
+        const extensionPage = windowHandles[0];
+        await driver.switchToWindow(extensionPage);
+        await driver.delay(500);
+
+        // click on the account menu icon
+        await driver.clickElement('.account-menu__icon');
+        await driver.delay(500);
+
+        // try to click on the notification item
+        await driver.clickElement({
+          text: 'Settings',
+          tag: 'div',
+        });
+        await driver.delay(500);
+
+        // try to click on the snaps item
+        await driver.clickElement({
+          text: 'Snaps',
+          tag: 'div',
+        });
+        await driver.delay(500);
+
+        // try to disable the snap
+        await driver.clickElement('.toggle-button > div');
+
+        // switch back to test-snaps window
+        await driver.waitUntilXWindowHandles(2, 5000, 10000);
+        windowHandles = await driver.getAllWindowHandles();
+        await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
+
+        // wait then try the notification test
+        await driver.delay(1000);
+        await driver.clickElement('#sendInAppNotification');
+
+        // click OK on the popup
+        await driver.delay(1000);
+        // driver.switchTo().alert().accept();
+        await driver.closeAlertPopup();
+        await driver.delay(1000);
+
+        // switch back to snaps page
+        await driver.switchToWindow(extensionPage);
+        await driver.delay(500);
+
+        // try to re-enaable the snap
+        await driver.clickElement('.toggle-button > div');
+        await driver.delay(500);
+
+        // switch back to test snaps page
+        await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
+
+        // wait then try the notification test
+        await driver.delay(1000);
+        await driver.clickElement('#sendInAppNotification');
+
+        // check to see that there is one notification
+        await driver.switchToWindow(extensionPage);
+        await driver.delay(500);
+        const notificationResult = await driver.findElement(
+          '.account-menu__icon__notification-count',
+        );
+        assert.equal(await notificationResult.getText(), '1');
+
+        // click on see details
+        await driver.clickElement({ text: 'See details', tag: 'button' });
+        await driver.delay(500);
+
+        // try to remove snap
+        await driver.clickElement({ text: 'Remove Snap', tag: 'button' });
+        await driver.delay(500);
+
+        // try to click remove on popover
+        await driver.clickElement({
+          xpath: '/html/body/div[2]/div/div/section/div[3]/button[2]',
```

Comment:
I'm not sure I understand your reasoning for this 🤔

Reply:
As I stated above, the second, popped-over "remove snap" button was unadressable through any means other than xpath here. Selenium claims that the popover was hijacking the click (rather than just accepting the click) and even though we try to click on said popover. That makes the test fail.

Reply:
this is fixed with f6aaf81